04/app.js

```diff
@@ -1,8 +1,56 @@
-console.log('DOM');
+console.log("DOM");
 
 // struktura do wykorzystania w pętli
 const menuItems = [
-    {text: 'start', url: '/'},
-    {text: 'galeria', url: '/gallery'},
-    {text: 'kontakt', url: '/contact'},
-];
\ No newline at end of file
+	{ text: "start", url: "/" },
+	{ text: "galeria", url: "/gallery" },
+	{ text: "kontakt", url: "/contact" },
+];
+
+const navElement = document.querySelector("nav");
+const ulWrapper = document.createElement("ul");
+
+navElement.appendChild(ulWrapper);
+
+for (let i = 0; i < menuItems.length; i++) {
+	liElement = document.createElement("li");
+	linkEl = document.createElement("a");
+	ulWrapper.appendChild(liElement);
+	liElement.appendChild(linkEl);
+}
+
+const linksList = ulWrapper.querySelectorAll("a");
+
+// console.log(linksList);
+
+linksList.forEach(function (item, index) {
+	item.setAttribute("href", menuItems[index]["url"]);
+	item.innerText = menuItems[index]["text"];
```

Comment:
Tą część spokojnie można wrzucić do pętli `for` i wtedy nasz kod jest 2 praktycznie razy szybcy bo nie mamy 2 pętli a jedną ;)